openupgrade_scripts/scripts/project/16.0.1.2/upgrade_analysis_work.txt

```diff
@@ -0,0 +1,99 @@
+---Models in module 'project'---
+obsolete model project.delete.wizard [transient]
+# NOTHING TO DO: wizard removed
+
+---Fields in module 'project'---
+project      / account.analytic.tag     / task_ids (many2many)          : DEL relation: project.task
+project      / project.task             / analytic_tag_ids (many2many)  : DEL relation: account.analytic.tag
+# NOTHING TO DO: lost feature
+
+project      / project.milestone        / task_ids (one2many)           : NEW relation: project.task
+project      / project.project          / allow_milestones (boolean)    : NEW hasdefault: default
```

Comment:
The default of this one is not sure to be `True`, so better to initialize the column manually with True.

Reply:
I've tested a bit. The default value is set to conform to the current value of the `project.group_project_milestone`.

On a fresh install of `project` in 16.0, the milestone option is set to False. Which leads to a default value set to False when a project is created.

With demo data in 16.0 the milestone option is set to True, which leads to projects with a default value set to True.

Doing nothing in migration script leads to a null value for the `allow_milestone` column in database.

I made a proposal of setting the `allow_milestone` field to False in pre-migration script. It seams to work correctly with demo data also.

Reply:
I don't think that should be the analysis. The question is that on v15, there's no check for choosing if showing milestones or not, so a user always sees them all. So if you migrate from v15, you should mark it as True for preserving that the milestones are shown for all the existing projects. Later on, you will be able to unmark some of them if you want to, or the default will apply according the group.

Do you agree with my reasoning?

Reply:
Ok, I understand your analysis. I agree with you.

Reply:
Done, `allow_milestones` is created and set to True in pre-migration script.